Yosemite/Yosemite/Stores/StatsStoreV4.swift

```diff
@@ -82,6 +90,30 @@ public extension StatsStoreV4 {
             onCompletion(nil)
         }
     }
+
+    /// Retrieves the site visit stats associated with the provided Site ID (if any!).
+    ///
+    func retrieveSiteVisitStats(siteID: Int,
+                                timeRange: StatsTimeRangeV4,
+                                latestDateToInclude: Date,
+                                onCompletion: @escaping (Error?) -> Void) {
+
+        let remote = SiteVisitStatsRemote(network: network)
+
+        remote.loadSiteVisitorStats(for: siteID,
+                                    unit: timeRange.siteVisitStatsUnitGranularity,
+                                    latestDateToInclude: latestDateToInclude,
+                                    quantity: 1) { [weak self] (siteVisitStats, error) in
+                                        guard let siteVisitStats = siteVisitStats else {
+                                            onCompletion(error.flatMap({ SiteVisitStatsStoreError(error: $0) }))
```

Comment:
This is interesting. Is the API returning multiple error codes or possibly duplicates? The `Remote.swift` class shows only a single error should return, not an array.

Reply:
actually I'll update the `flatMap` to `map` since the `SiteVisitStatsStoreError` initializer is not optional. It's a single generic error from the API, and here we map the generic error to `SiteVisitStatsStoreError` error type for the app to handle each site visit stats specific case